pydantic/env_settings.py

```diff
@@ -244,3 +240,16 @@ def read_env_file(
         return {k.lower(): v for k, v in file_vars.items()}
     else:
         return file_vars
+
+
+def read_secret_path(secrets_path: Path, env_name: str, case_sensitive: bool = False) -> Optional[str]:
+    paths = [secrets_path / env_name] if case_sensitive else [secrets_path / env_name, secrets_path / env_name.upper()]
```

Comment:
This doesn't work if the filename is partially uppercase, e.g. `SECRET_var`.

To fix this, I think you'll need a utility function like 

```py
def find_case_insensitive_path(path: Path) -> Path:
    """
    Find a file within path's directory with a case-insensitive matching filename.
    
    Note: this won't if the rest of the path has a case mismatch.
    """
    expected_name = path.name.lower()
    for f in path.parent.iterdir():
        if f.name.lower() == expected_name:
            return f
    return path
```

Then you can reverse your changes in `__call__` above, and just add

```py
path = secrets_path / env_name
if settings.__config__.case_sensitive not path.exists():
    path = find_case_insensitive_path(path)
```

Reply:
Thanks for the feedback :) 

But, we have two scenarios:

- Case sensitive and the filename which has the same syntax
- Case insensitive and read files with linux (issues #3304)

Reply:
not sure what you mean by "same syntax"?

Reply:
Same syntax for the name of files (for example when you have a file in upper case if you are case sensitive you should not read it if the variable name is in lower case)